src/Umbraco.Cms.Integrations.Crm.Dynamics/App_Plugins/UmbracoCms.Integrations/Crm/Dynamics/js/formpicker.controller.js

```diff
@@ -77,6 +80,7 @@
                 });
             }
         });
+        vm.loading = false;
```

Comment:
Looks like this should be within the callback function, not outside it, otherwise it will execute immediately.